src/HotChocolate/Core/src/Types/Utilities/Serialization/InputObjectCompiler.cs

```diff
@@ -164,7 +165,21 @@ private static Expression GetFieldValue(InputField field, Expression fieldValues
     }
 
     private static Dictionary<string, InputField> CreateFieldMap(InputObjectType type)
-        => type.Fields.ToDictionary(t => t.Property!.Name, StringComparer.Ordinal);
+    {
```

Comment:
I am still weighing the changes in my head :) 

It's a big breaking change.

Your implementation looks now clean.

Reply:
I think I will take another day to reflect on this and run the tests.

Also, can you introduce some tests that show that the new behavior works?

Great work so far.

Reply:
@michaelstaib I've added a test to show the new behaviour working.

Reply:
Awesome... thanks for your patience and work on this!